packages/core/src/external/carequality/iti55/validating-iti55.ts

```diff
@@ -0,0 +1,176 @@
+import {
+  Patient as FHIRPatient,
+  Address as FHIRAddress,
+  ContactPoint,
+  Identifier,
+} from "@medplum/fhirtypes";
+import { USState, getStateEnum } from "../../../domain/geographic-locations";
+import {
+  PatientDataMPI,
+  Address,
+  Contact,
+  PersonalIdentifier,
+} from "../../mpi/patient-incoming-schema";
+
+export class PatientAddressRequestedError extends Error {
+  constructor(message?: string) {
+    super(message);
+    this.name = "PatientAddressRequestedError";
+  }
+}
+
+export class LivingSubjectAdministrativeGenderRequestedError extends Error {
+  constructor(message?: string) {
+    super(message);
+    this.name = "LivingSubjectAdministrativeGenderRequestedError";
+  }
+}
+
+export class InternalError extends Error {
+  constructor(message?: string) {
+    super(message);
+    this.name = "InternalError";
+  }
+}
+
+export function validateFHIRAndExtractPatient(patient: FHIRPatient): PatientDataMPI {
+  const firstName = patient.name?.[0]?.given?.[0]; // TODO we are taking the first index here but there might be multiple given names
+  if (!firstName) {
+    throw new InternalError("Given name is not defined");
+  }
+  const lastName = patient.name?.[0]?.family;
+  if (!lastName) {
+    throw new InternalError("Family name is not defined");
+  }
+  const birthDate = patient.birthDate;
+  if (!birthDate) {
+    throw new InternalError("Birth date is not defined");
+  }
+  const genderAtBirth = patient.gender;
```

Comment:
If someone is querying us and they are missing multiple required fields, they will only know about the first one they are missing in the error message. I think that is ok, but acknowledging that behavior